agent/consul/state/catalog_schema.go

```diff
@@ -0,0 +1,355 @@
+package state
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/hashicorp/go-memdb"
+
+	"github.com/hashicorp/consul/agent/structs"
+)
+
+const (
+	tableNodes    = "nodes"
+	tableServices = "services"
+	tableChecks   = "checks"
+
+	indexID               = "id"
+	indexServiceName      = "service"
+	indexConnect          = "connect"
+	indexKind             = "kind"
+	indexStatus           = "status"
+	indexNodeServiceCheck = "node_service_check"
+	indexNodeService      = "node_service"
```

Comment:
These constants should make it much easier to navigate between queries that use an index, and the index definition. It also makes it easy to find all queries that use a particular index.